ini/stm32h7.ini

```diff
@@ -137,3 +137,29 @@ board                       = marlin_STM32H723ZE
 [env:STM32H723ZG_btt]
 extends                     = STM32H723Zx_btt
 board                       = marlin_STM32H723ZG
+
+#
+# FYSETC Spider King v1.1 (STM32H723ZGT6 ARM Cortex-M7)
+#
+[env:FYSETC_SPIDER_KING723]
+board                       = marlin_STM32H723ZG
+extends                     = stm32_variant
+platform                    = ststm32@15.4.1
+platform_packages           = framework-arduinoststm32@~4.20200.220530
+board_build.offset          = 0x20000
+board_upload.offset_address = 0x08020000
+build_flags                 = ${stm32_variant.build_flags}
+                              -DPIN_SERIAL1_TX=PA_9 -DPIN_SERIAL1_RX=PA_10
+                              -DPIN_SERIAL2_TX=PD_5 -DPIN_SERIAL2_RX=PD_6
+                              -DPIN_SERIAL3_TX=PD_8 -DPIN_SERIAL3_RX=PD_9
+                              -DPIN_SERIAL4_TX=PA_0 -DPIN_SERIAL4_RX=PA_1
+                              -DPIN_SERIAL7_TX=PE_8 -DPIN_SERIAL7_RX=PE_7
+                              -DSERIAL_RX_BUFFER_SIZE=1024 -DSERIAL_TX_BUFFER_SIZE=1024
+                              -DTIMER_SERVO=TIM5 -DTIMER_TONE=TIM2
+                              -DSTEP_TIMER_IRQ_PRIO=0
+                              -DRCC_PERIPHCLK_I2C35=RCC_PERIPHCLK_I2C5
+                              -DUSE_USB_HS -DUSE_USB_HS_IN_FS
+                              -DD_CACHE_DISABLED
+                              -UI2C5_BASE
+upload_protocol             = cmsis-dap
+debug_tool                  = cmsis-dap
```

Comment:
I just noticed that this is nearly an exact copy of `[STM32H723Zx_btt]` / BigTreeTech STM32H723Zx common environment.

Are all these build flags applicable? Was this PR tested on real hardware?

Reply:
Thank you for your excellent work, which has been of great help to me. Just now, I verified the modified code.

A. After modification, the code:```Board-INFO-NAME" Spider King V1.1```may not be displayed fully under 12864, so I suggest changing it to``` Board-INFO-NAME "SpiderKing V1.1``` to display all content normally.

B. There seems to be an issue with using I2C EEPROM. After enabling EEPROM, I used m501 and m500:```echo: EEPROM version mismatch (EEPROM=?"? "Marlin=V90)```, add the following code under pins-FYSETC-SPIDER_KING723. h: 

```
#if ANY(NO_EEPROM_SELECTED, FLASH_EEPROM_EMULATION)
  #undef NO_EEPROM_SELECTED
  #ifndef FLASH_EEPROM_EMULATION
    #define FLASH_EEPROM_EMULATION
  #endif
  #define EEPROM_PAGE_SIZE      (0x800UL) // 2K
  #define EEPROM_START_ADDRESS  (0x8000000UL + (STM32_FLASH_SIZE) * 1024UL - (EEPROM_PAGE_SIZE) * 2UL)
  #define MARLIN_EEPROM_SIZE    EEPROM_PAGE_SIZE
#endif
```

can read and write eproom normally, The issue with i2c may require your assistance.

C. "[STM32H723Zx_btt] is similar. In fact, STM32H723VGT6 is also similar to this configuration and can be reused. However, because it is STM32H723Zx_btt, I cannot use it directly. Perhaps I can change it:

```
[STM32H723]
extends                     = stm32_variant
platform                    =  ststm32@15.4.1
platform_packages           = framework-arduinoststm32@~4.20200.220530
board_build.offset          = 0x20000
board_upload.offset_address = 0x08020000
Build_flags=${stm32_variant. build_flags}
-DPIN-SERIAL1TX=PA_9- DPIN-SERIAL1RX=PA_10
-DPIN-SERIAL2TX=PD_5- DPIN-SERIAL2RX=PD_6
-DPIN-SERIAL3-TX=PD_8- DPIN-SERIAL3-RX=PD_9
-DPIN-SERIAL4-TX=PA-0- DPIN-SERIAL4-RX=PA_1
-DPIN-SERIAL7-TX=PE_8- DPIN-SERIAL7-RX=PE_7
-DSERIAL-RX-BUFFER-SIZE=1024- DSERIAL-TX-BUFFER-SIZE=1024
-DTIMER_SERVO=TIM5- DTIMER-TONE=TIM2
-DSTEP_TIMER-IRQ-PRIO=0
-DRCC-PERIPHCLKI2C35=RCCPERIPHCLKI2C5
-DUSE-USB-HS - DUSE-USB-HS.IN-FS
-DD-CACHE-DISABLED
-UI2C5-BASE
upload_protocol             = cmsis-dap
debug_tool                  = cmsis-dap

[env: FYSETL_SPIDER_KING723]
extends                     = STM32H723
board                       = marlin_STM32H723ZG

[env: FYSETL_SPIDER_30]
extends                     = STM32H723
board                       = marlin_STM32H723VG

......
```


It would be better, of course, this is just a suggestion.
![微信图片_20240527162523](https://github.com/MarlinFirmware/Marlin/assets/145735142/97c2b875-dd28-4925-8262-1d4b1fe37682)

Reply:
> After modification, the code:`Board-INFO-NAME" Spider King V1.1`may not be displayed fully under 12864, so I suggest changing it to` Board-INFO-NAME "SpiderKing V1.1` to display all content normally.

> <img width="483" alt="image" src="https://github.com/MarlinFirmware/Marlin/assets/13375512/be458c13-48f7-4bd4-857b-7b428a07e7ff">

Since you didn't set a `CUSTOM_MACHINE_NAME`, Marlin is defaulting it to `DEFAULT_MACHINE_NAME` which defaults to `BOARD_INFO_NAME` when not defined.

As long as `BOARD_INFO_NAME` is 20 characters or less, that'll work fine under the "About Printer" menu, but you should be setting your own `CUSTOM_MACHINE_NAME` so it shows up correctly on the Status screen.

> There seems to be an issue with using I2C EEPROM. After enabling EEPROM, I used m501 and m500:echo: EEPROM version mismatch (EEPROM=?"? "Marlin=V90)

Hmm. Maybe FYSETC removed the external EEPROM in later Spider King versions? @GerogeFu?

Reply:
Thank you for your help. Yes, just make the necessary changes to enable it：

```
// Name displayed in the LCD "Ready" message and Info menu
#define CUSTOM_MACHINE_NAME "3D Printer"
```

will not be displayed when defining CUSTOM-Machine-NAME, but when I haven't set it up, I also hope it can display the complete motherboard information. Okay, I'm a bit obsessive-compulsive.

> Hmm. Maybe FYSETC removed the external EEPROM in later Spider King versions?

FYSETC did not remove Spider King external eproom, which can be used under the F4 chip, but I cannot use it after replacing it with the H7 chip. Perhaps there are other reasons.

I'm not GerogeFu.

Reply:
> FYSETC did not remove Spider King external eproom, which can be used under the F4 chip, but I cannot use it after replacing it with the H7 chip

This is where things can get complicated since you can technically have a Spider King V1.0 or V1.1 with STM32F4 or STM32H7 MCUs. This means a working EEPROM implementation isn't tied to the specific board, but board + MCU combination.

I'll have to think of how that fits into Marlin's way of handling board versions now that we have that MCU variable.

> I'm not GerogeFu.

They are the FYSETC point of contact and have contributed to the project, so I was hoping they would have more information.